tests/lib/rules/no-unused-services.js

```diff
@@ -0,0 +1,86 @@
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-unused-services');
+const RuleTester = require('eslint').RuleTester;
+
+const { ERROR_MESSAGE } = rule;
+const message = ERROR_MESSAGE('fooName');
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+  },
+  parser: require.resolve('babel-eslint'),
+});
+
+const usecases = [
+  'this.fooName;',
+  'this.fooName.prop;',
+  'this.fooName.func();',
+  "this.get('fooName');",
+  "this.get('fooName.prop');",
+  "get(this, 'fooName');",
+  "get(this, 'fooName.prop');",
+  "this.getProperties('a', 'fooName');",
+  "this.getProperties('a', 'fooName.prop');",
+  "this.getProperties(['a', 'fooName']);",
+  "this.getProperties(['a', 'fooName.prop']);",
+  "getProperties(this, 'a', 'fooName');",
+  "getProperties(this, 'a', 'fooName.prop');",
+  "getProperties(this, ['a', 'fooName']);",
+  "getProperties(this, ['a', 'fooName.prop']);",
+  'const { a, b, fooName } = this;',
+  'let { a, fooName, b } = this;',
+];
+
+const valid = [];
+for (const use of usecases) {
+  valid.push({
+    code: `
+      class MyClass {
+        @service('foo')
+        fooName;
+
+        fooFunc() { ${use} }
+      }
+    `,
+    parser: require.resolve('babel-eslint'),
+  });
+
+  valid.push({
+    code: `
+      Component.extend({
+        fooName: service('foo'),
+
+        fooFunc() { ${use} }
+      });
+    `,
+    parser: require.resolve('babel-eslint'),
+  });
+}
+
+ruleTester.run('no-unused-services', rule, {
+  valid,
+  invalid: [
+    {
+      code: "class MyClass { @service('foo') fooName; }",
```

Comment:
We'll want to include some other property accesses in this class to ensure we don't have false positives from accessing unrelated properties. One way you could do this is to turn the `useCases` array into a function which generates the list of use cases with a parameterized property name:

```js
class MyClass { @service('foo') fooName; makeThisPropertyAccesses('unrelatedProperty').join('; '); }
```

Reply:
Great suggestion~ added!